nano/node/blockprocessor.cpp

```diff
@@ -48,6 +48,12 @@ bool nano::block_processor::full ()
 	return (blocks.size () + state_blocks.size ()) > node.flags.block_processor_full_size;
 }
 
+bool nano::block_processor::half_full ()
+{
+	std::unique_lock<std::mutex> lock (mutex);
+	return (blocks.size () + state_blocks.size ()) > node.flags.block_processor_full_size / 2;
```

Comment:
Could extract `(blocks.size () + state_blocks.size ())`  into a `size ()` method since it's repeated in `full ()`